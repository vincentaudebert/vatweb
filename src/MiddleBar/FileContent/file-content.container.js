import { connect } from 'react-redux';
import FileContent from './file-content';

const mapStateToProps = (state, ownProps) => {
  return {
    openFiles: state.openFiles,
  };
};

const mapDispatchToProps = (dispatch, ownProps) => ({});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(FileContent);
