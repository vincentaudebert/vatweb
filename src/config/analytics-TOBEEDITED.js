export default 'UA-XXXXXXX-X';
