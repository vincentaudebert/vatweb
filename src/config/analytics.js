// @flow
const getGaId = (): string => {
  let gaId = 'UA-XXXXXXX-X';
  if (process.env.GOOGLE_ANALYTICS_ID) {
    gaId = process.env.GOOGLE_ANALYTICS_ID;
  }
  return gaId;
};

export default getGaId;
