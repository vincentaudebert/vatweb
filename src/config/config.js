import APOCOPE from './cv/experience/1-apocope';
import ALLIANZ from './cv/experience/2-allianz';
import SHOWCASE_WORKSHOP from './cv/experience/3-showcase-workshop';
import PIKPOK from './cv/experience/4-pikpok';
import SPRINGLOAD from './cv/experience/5-springload';
import SMOOTHCODE from './cv/experience/6-smooth-code';
import EDUCATION from './cv/education';
import HOBBIES from './cv/hobbies';
import ME from './cv/me';
import WELCOME from './cv/welcome';

export default [
  {
    name: 'experience',
    type: 'folder',
  },
  {
    name: '1-apocope.js',
    content: APOCOPE,
    type: 'file',
    language: 'react',
    inFolder: true,
  },
  {
    name: '2-allianz.js',
    content: ALLIANZ,
    type: 'file',
    language: 'react',
    inFolder: true,
  },
  {
    name: '3-showcase-workshop.js',
    content: SHOWCASE_WORKSHOP,
    type: 'file',
    language: 'react',
    inFolder: true,
  },
  {
    name: '4-pikpok.js',
    content: PIKPOK,
    type: 'file',
    language: 'react',
    inFolder: true,
  },
  {
    name: '5-springload.js',
    content: SPRINGLOAD,
    type: 'file',
    language: 'react',
    inFolder: true,
  },
  {
    name: '6-smooth-code.js',
    content: SMOOTHCODE,
    type: 'file',
    language: 'react',
    inFolder: true,
  },
  {
    name: 'education.js',
    content: EDUCATION,
    type: 'file',
    language: 'react',
    inFolder: false,
  },
  {
    name: 'hobbies.js',
    content: HOBBIES,
    type: 'file',
    language: 'react',
    inFolder: false,
  },
  {
    name: 'me.js',
    content: ME,
    type: 'file',
    language: 'react',
    inFolder: false,
  },
  {
    name: 'welcome.json',
    content: WELCOME,
    type: 'file',
    language: 'json',
    inFolder: false,
  },
];
