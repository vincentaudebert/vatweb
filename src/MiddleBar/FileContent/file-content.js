import React from 'react';
import Tabs from '../Tabs/tabs.container';
import SyntaxHighlighter, {
  registerLanguage,
} from 'react-syntax-highlighter/prism-light';
import jsx from 'react-syntax-highlighter/languages/prism/jsx';
import prism from 'react-syntax-highlighter/styles/prism/atom-dark';
import config from '../../config/config';
import { getActiveItem, getFromSlug } from '../../utilities';

const Component = props => {
  const activeFile = getActiveItem(props.openFiles);
  const fullActiveFile = activeFile
    ? getFromSlug(config, activeFile.name)
    : false;
  const codeString = fullActiveFile ? fullActiveFile.content : '';

  registerLanguage('jsx', jsx);
  return (
    <div>
      <Tabs />
      <div className="file-content">
        <SyntaxHighlighter showLineNumbers language="jsx" style={prism}>
          {codeString}
        </SyntaxHighlighter>
      </div>
    </div>
  );
};

export default Component;
