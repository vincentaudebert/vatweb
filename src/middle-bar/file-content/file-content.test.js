import React from 'react';
import renderer from 'react-test-renderer';
import FileContent from './file-content';
import config from '../../config/config';

jest.mock('../tabs/tabs.container', () => 'div');
jest.mock('react-syntax-highlighter/dist/esm/styles/prism', () => ({
  atomDark: undefined,
}));

describe('FileContent', () => {
  let currentFile;

  beforeEach(() => {
    [currentFile] = config.filter(item => item.name === 'welcome.json');
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<FileContent currentFile={currentFile} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with undefined file', () => {
    const tree = renderer
      .create(<FileContent currentFile={undefined} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
