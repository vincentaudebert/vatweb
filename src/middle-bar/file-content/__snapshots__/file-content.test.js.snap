// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileContent renders correctly 1`] = `
<div>
  <div />
  <div
    className="file-content"
  >
    <pre
      style={
        Object {
          "MozHyphens": "none",
          "MozTabSize": "4",
          "OTabSize": "4",
          "WebkitHyphens": "none",
          "background": "#1d1f21",
          "borderRadius": "0.3em",
          "color": "#c5c8c6",
          "direction": "ltr",
          "fontFamily": "Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace",
          "hyphens": "none",
          "lineHeight": "1.5",
          "margin": ".5em 0",
          "msHyphens": "none",
          "overflow": "auto",
          "padding": "1em",
          "tabSize": "4",
          "textAlign": "left",
          "textShadow": "0 1px rgba(0, 0, 0, 0.3)",
          "whiteSpace": "pre",
          "wordBreak": "normal",
          "wordSpacing": "normal",
        }
      }
    >
      <code
        style={
          Object {
            "MozHyphens": "none",
            "MozTabSize": "4",
            "OTabSize": "4",
            "WebkitHyphens": "none",
            "color": "#c5c8c6",
            "direction": "ltr",
            "float": "left",
            "fontFamily": "Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace",
            "hyphens": "none",
            "lineHeight": "1.5",
            "msHyphens": "none",
            "paddingRight": "10px",
            "tabSize": "4",
            "textAlign": "left",
            "textShadow": "0 1px rgba(0, 0, 0, 0.3)",
            "whiteSpace": "pre",
            "wordBreak": "normal",
            "wordSpacing": "normal",
          }
        }
      >
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          1
          
        </span>
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          2
          
        </span>
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          3
          
        </span>
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          4
          
        </span>
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          5
          
        </span>
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          6
          
        </span>
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          7
          
        </span>
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          8
          
        </span>
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          9
          
        </span>
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          10
          
        </span>
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          11
          
        </span>
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          12
          
        </span>
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          13
          
        </span>
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          14
          
        </span>
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          15
          
        </span>
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          16
          
        </span>
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          17
          
        </span>
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          18
          
        </span>
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          19
          
        </span>
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          20
          
        </span>
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          21
          
        </span>
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          22
          
        </span>
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          23
          
        </span>
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          24
          
        </span>
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          25
          
        </span>
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          26
          
        </span>
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          27
          
        </span>
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          28
          
        </span>
      </code>
      <code
        style={
          Object {
            "MozHyphens": "none",
            "MozTabSize": "4",
            "OTabSize": "4",
            "WebkitHyphens": "none",
            "color": "#c5c8c6",
            "direction": "ltr",
            "fontFamily": "Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace",
            "hyphens": "none",
            "lineHeight": "1.5",
            "msHyphens": "none",
            "tabSize": "4",
            "textAlign": "left",
            "textShadow": "0 1px rgba(0, 0, 0, 0.3)",
            "whiteSpace": "pre",
            "wordBreak": "normal",
            "wordSpacing": "normal",
          }
        }
      >
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          {
        </span>
        
          
        <span
          className={undefined}
          style={
            Object {
              "color": "#A8FF60",
            }
          }
        >
          "hello"
        </span>
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          :
        </span>
         
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          [
        </span>
        
            
        <span
          className={undefined}
          style={
            Object {
              "color": "#A8FF60",
            }
          }
        >
          "Heya! Welcome on my portfolio."
        </span>
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          ,
        </span>
         
            
        <span
          className={undefined}
          style={
            Object {
              "color": "#A8FF60",
            }
          }
        >
          "It shows you what I have done and what I can do."
        </span>
        
          
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          ]
        </span>
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          ,
        </span>
        
          
        <span
          className={undefined}
          style={
            Object {
              "color": "#A8FF60",
            }
          }
        >
          "wait-what"
        </span>
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          :
        </span>
         
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          [
        </span>
        
            
        <span
          className={undefined}
          style={
            Object {
              "color": "#A8FF60",
            }
          }
        >
          "Ok, maybe it needs a bit of explanation."
        </span>
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          ,
        </span>
        
            
        <span
          className={undefined}
          style={
            Object {
              "color": "#A8FF60",
            }
          }
        >
          "You can use the file explorer on your left to open files"
        </span>
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          ,
        </span>
        
            
        <span
          className={undefined}
          style={
            Object {
              "color": "#A8FF60",
            }
          }
        >
          "These describe some of the work I have been doing or what studies I did, etc."
        </span>
        
          
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          ]
        </span>
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          ,
        </span>
        
          
        <span
          className={undefined}
          style={
            Object {
              "color": "#A8FF60",
            }
          }
        >
          "are-you-mad"
        </span>
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          :
        </span>
         
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          [
        </span>
        
            
        <span
          className={undefined}
          style={
            Object {
              "color": "#A8FF60",
            }
          }
        >
          "Hehe, no. I just thought it would be"
        </span>
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          ,
        </span>
        
            
        <span
          className={undefined}
          style={
            Object {
              "color": "#A8FF60",
            }
          }
        >
          "an original way to present myself and relevant with my expertise."
        </span>
        
          
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          ]
        </span>
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          ,
        </span>
        
          
        <span
          className={undefined}
          style={
            Object {
              "color": "#A8FF60",
            }
          }
        >
          "aint-nobody-got-time-for-that"
        </span>
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          :
        </span>
         
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          [
        </span>
        
            
        <span
          className={undefined}
          style={
            Object {
              "color": "#A8FF60",
            }
          }
        >
          "Ok if you can't be bothered"
        </span>
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          ,
        </span>
        
            
        <span
          className={undefined}
          style={
            Object {
              "color": "#A8FF60",
            }
          }
        >
          "you can download directly my CV in a PDF format:"
        </span>
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          ,
        </span>
        
            
        <span
          className={undefined}
          style={
            Object {
              "color": "#A8FF60",
            }
          }
        >
          "https://github.com/vincentaudebert/vatweb/raw/master/public/assets/Vincent_Audebert_CV.pdf"
        </span>
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          ,
        </span>
        
            
        <span
          className={undefined}
          style={
            Object {
              "color": "#A8FF60",
            }
          }
        >
          "or click on the PDF icon in the menu bar on the left (desktop only)"
        </span>
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          ,
        </span>
        
            
        <span
          className={undefined}
          style={
            Object {
              "color": "#A8FF60",
            }
          }
        >
          "or click on the smiley face in the footer bar."
        </span>
        
          
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          ]
        </span>
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          ,
        </span>
        
          
        <span
          className={undefined}
          style={
            Object {
              "color": "#A8FF60",
            }
          }
        >
          "where-is-da-code"
        </span>
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          :
        </span>
         
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          [
        </span>
        
            
        <span
          className={undefined}
          style={
            Object {
              "color": "#A8FF60",
            }
          }
        >
          "Oh by the way, I made the source code open source."
        </span>
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          ,
        </span>
        
            
        <span
          className={undefined}
          style={
            Object {
              "color": "#A8FF60",
            }
          }
        >
          "You can see it here:"
        </span>
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          ,
        </span>
        
            
        <span
          className={undefined}
          style={
            Object {
              "color": "#A8FF60",
            }
          }
        >
          "https://github.com/vincentaudebert/vatweb"
        </span>
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          ,
        </span>
        
            
        <span
          className={undefined}
          style={
            Object {
              "color": "#A8FF60",
            }
          }
        >
          "or click on the git icon in the footer bar."
        </span>
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          ,
        </span>
        
          
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          ]
        </span>
        
        
        <span
          className={undefined}
          style={
            Object {
              "color": "#c5c8c6",
            }
          }
        >
          }
        </span>
      </code>
    </pre>
  </div>
</div>
`;

exports[`FileContent renders correctly with undefined file 1`] = `
<div>
  <div />
  <div
    className="file-content"
  >
    <pre
      style={
        Object {
          "MozHyphens": "none",
          "MozTabSize": "4",
          "OTabSize": "4",
          "WebkitHyphens": "none",
          "background": "#1d1f21",
          "borderRadius": "0.3em",
          "color": "#c5c8c6",
          "direction": "ltr",
          "fontFamily": "Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace",
          "hyphens": "none",
          "lineHeight": "1.5",
          "margin": ".5em 0",
          "msHyphens": "none",
          "overflow": "auto",
          "padding": "1em",
          "tabSize": "4",
          "textAlign": "left",
          "textShadow": "0 1px rgba(0, 0, 0, 0.3)",
          "whiteSpace": "pre",
          "wordBreak": "normal",
          "wordSpacing": "normal",
        }
      }
    >
      <code
        style={
          Object {
            "MozHyphens": "none",
            "MozTabSize": "4",
            "OTabSize": "4",
            "WebkitHyphens": "none",
            "color": "#c5c8c6",
            "direction": "ltr",
            "float": "left",
            "fontFamily": "Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace",
            "hyphens": "none",
            "lineHeight": "1.5",
            "msHyphens": "none",
            "paddingRight": "10px",
            "tabSize": "4",
            "textAlign": "left",
            "textShadow": "0 1px rgba(0, 0, 0, 0.3)",
            "whiteSpace": "pre",
            "wordBreak": "normal",
            "wordSpacing": "normal",
          }
        }
      >
        <span
          className="react-syntax-highlighter-line-number"
          style={Object {}}
        >
          1
          
        </span>
      </code>
      <code
        style={
          Object {
            "MozHyphens": "none",
            "MozTabSize": "4",
            "OTabSize": "4",
            "WebkitHyphens": "none",
            "color": "#c5c8c6",
            "direction": "ltr",
            "fontFamily": "Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace",
            "hyphens": "none",
            "lineHeight": "1.5",
            "msHyphens": "none",
            "tabSize": "4",
            "textAlign": "left",
            "textShadow": "0 1px rgba(0, 0, 0, 0.3)",
            "whiteSpace": "pre",
            "wordBreak": "normal",
            "wordSpacing": "normal",
          }
        }
      />
    </pre>
  </div>
</div>
`;
